Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -1825,27 +1829,6 @@ theorem inner_sum_smul_sum_smul_of_sum_eq_zero {ι₁ : Type*} {s₁ : Finset ι
     mul_zero, Finset.sum_const_zero, zero_add, zero_sub, Finset.mul_sum, neg_div,
     Finset.sum_div, mul_div_assoc, mul_assoc]
 
-variable (𝕜)
-
-/-- `innerSL` is an isometry. Note that the associated `LinearIsometry` is defined in
-`InnerProductSpace.Dual` as `toDualMap`. -/
-@[simp]
-theorem innerSL_apply_norm (x : E) : ‖innerSL 𝕜 x‖ = ‖x‖ := by
-  refine
-    le_antisymm ((innerSL 𝕜 x).opNorm_le_bound (norm_nonneg _) fun y => norm_inner_le_norm _ _) ?_
-  rcases eq_or_ne x 0 with (rfl | h)
-  · simp
-  · refine (mul_le_mul_right (norm_pos_iff.2 h)).mp ?_
-    calc
-      ‖x‖ * ‖x‖ = ‖(⟪x, x⟫ : 𝕜)‖ := by
-        rw [← sq, inner_self_eq_norm_sq_to_K, norm_pow, norm_ofReal, abs_norm]
-      _ ≤ ‖innerSL 𝕜 x‖ * ‖x‖ := (innerSL 𝕜 x).le_opNorm _
-
-lemma norm_innerSL_le : ‖innerSL 𝕜 (E := E)‖ ≤ 1 :=
-  ContinuousLinearMap.opNorm_le_bound _ zero_le_one (by simp)
-
-variable {𝕜}
-
 /-- When an inner product space `E` over `𝕜` is considered as a real normed space, its inner
 product satisfies `IsBoundedBilinearMap`.
```

Comment:
This theorem generalizes too after #17011

Reply:
It seems that there is a problem with `HasStrictFDerivAt` in #17011. I don't really understand why the changes affect it as `HasStrictFDerivAt` has all `NormedAddCommGroup` in the assumption... Should I wait until it is resolved, correct?

Reply:
I've pushed a fix to that PR

Reply:
I saw it, and I'm afraid I'm not yet skilled enough to understand the fix... Could you explain briefly what was the issue and how you worked it around?

Reply:
The issue is that Lean was previously able to guess various arguments, but after that change no longer is. This is usually because it guesses them via unification (matching up terms to fill in the gaps).

Generalizing to seminormed additive groups means that the general seminormed theorems when used in the less-general normed context are now instantiated with `NormedAddCommGroup.toSeminormedAddCommGroup` or `SeminormedRing.toSeminormedAddCommGroup` instead of `inst` or `NormedRing.toNormedAddCommGroup`, and for whatever reason this stifle's Lean's ability to guess.

The `(E := E)`-style fixes are telling Lean "don't guess, here's the answer". The `:)` fixes are telling Lean "please finalize all your guesses now, no looking outside the brackets to help as this might confuse you).

It's very plausible that there is a Lean bug hiding here, but correctly handling unification and synthesis is a very complex can of worms that isn't worth opening on small annoyances like the ones that came up here.

Reply:
Yes, you can see it by doing `#synth CompatibleSMul E E ℝ 𝕜` (in the context where it worked, before the variables were generalized).

I think the thing that `#synth` finds is one that relies on `T2Space E`, which is equivalent to the space being normed and not seminormed. I think I saw something analogous in my last commit, and the fix was `letI : IsScalarTower ℝ 𝕜 E := RestrictScalars.isScalarTower _ _ _` which amounts to "find a different instance elsewhere".

Reply:
I see, that command seems very useful, thank you!

However, if I put `#synth CompatibleSMul E E ℝ 𝕜` just below `IsSymmetric.restrictScalars`, I get `failed to synthesize SMul ℝ E`... I also get the same with the following MWE.

```lean4
import Mathlib

variable {𝕜 E : Type*} [RCLike 𝕜] [NormedAddCommGroup E] [InnerProductSpace 𝕜 E]

open RCLike
open ComplexConjugate
open LinearMap

#synth CompatibleSMul E E ℝ 𝕜
```